enhancements/installer/kubernete-slb-owned-api-loadbalancers.md

```diff
@@ -0,0 +1,508 @@
+---
+title: kubernetes-slb-owned-api-loadbalancers
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - "@sttts"
+  - "@smarterclayton"
+approvers:
+  - "@sttts"
+  - "@smarterclayton"
+creation-date: 2020-05-06
+last-updated: 2020-08-26
+status: implementable
+see-also:
+  -   
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Kubernetes Service Owned API Loadbalancers
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. Service type Loadbalancer's ExternalTrafficPolicy should be Cluster or Local ?
+2. How to reserve private IPs on AWS ?
+
+## Summary
+
+Currently the installer is responsible for creating the loadbalancers for API and API-INT endpoints for the cluster. These loadbalancers 
+are only created at install-time and their lifecycle is not owned by the cluster. Since the Kubernetes API is hosted in pods in the cluster, 
+using the Kubernetes service type `Loadbalancer` to manage the lifecycle of the loadbalancers is next-step for self-hosted Kubernetes.
+
+## Motivation
+
+### Goals
+
+1. The API, API-INT loadbalancers are managed by Kubernetes service type `Loadbalancer`.
+2. Recovery of the cluster from some, to *all* of the control-plane DOWN scenarios should not be substantially more difficult. We do not want 
+   to trade cluster owned lifecycle for easy operations for harder recovery of control-plane.
+4. Support this deployment model for all platforms that support service type `Loadbalancer`.
+5. Allow customers to opt-out for user-provisioned workflows.
```

Comment:
This is 100% a blocker for me.  There should be 1 supported configuration per platform.  If we're allowing UPI clusters and IPI clusters to drift, it creates a significant support burden.